### PR TITLE
Lst 712 fix login issues

### PR DESCRIPTION
**Pull request by markhigham**

### CODE REVIEW

**Comment by markhigham** [comment]
Hey chaps. I've updated the code so that eslint passes on codeclimate.

File: src/app/controllers/controller.login.js [hunk 1]
```diff
@@ -28,10 +29,26 @@ function createClearUserCookie(){
 	return createClearCookieStr( config.userCookie.name );
 }
 
+function convertDataCookieToLocalCookie(cookieFromData) {
+
+	const localCookie = cookieFromData
+		.split(";")
```
Comment (web-bert) [inline src/app/controllers/controller.login.js]: I believe the cookie spec is to separate with a semi colon and a space so this should be `.split('; ')` and then the join should be `.join('; ')` and   then you can remove the `const trimmed = part.trim()`

**Comment by jim68000** [comment]
110 fixed issues!

File: src/app/lib/get-redirect-uri.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const url = require('url');
+const config = require('../config');
+
+module.exports = function(req){
+	const protocol = config.isDev ? req.protocol : 'https';
+	
+	const callbackUrl = url.format({
+		protocol: protocol,
```
Comment (web-bert) [inline src/app/lib/get-redirect-uri.js]: You can replace the `protocol: protocol,` with just `protocol,`

File: src/app/controllers/controller.login.js (hunk 1 above)
Comment (markhigham) [reply src/app/controllers/controller.login.js]: good call. I'll do this.

File: src/app/lib/get-redirect-uri.js (hunk 2 above)
Comment (markhigham) [reply src/app/lib/get-redirect-uri.js]: I do not like this syntax!. But it fits with the rest of this solution. So I'll do it :grimacing:

Comment (web-bert) [reply src/app/lib/get-redirect-uri.js]: It's an optional thing :-)

Comment (markhigham) [reply src/app/lib/get-redirect-uri.js]: i also fear change

File: src/test/app/specs/controllers/controller.login.spec.js [hunk 3]
```diff
@@ -44,11 +47,16 @@ describe( 'Login controller', function(){
 			clearCookie: spy( 'res.clearCookie' )
 		};
 
+		getRedirectUri = function(){
```
Comment (web-bert) [inline src/test/app/specs/controllers/controller.login.spec.js]: This should be `getRedirectUri = spy( 'getRedirectUri', redirectUri );` and then there should be a test for `expect( getRedirectUri ).toHaveBeenCalledWith( req );`

Comment (markhigham) [reply src/test/app/specs/controllers/controller.login.spec.js]: :+1:

**Pull request merged**